libcaf_core/caf/flow/observable.test.cpp

```diff
@@ -438,50 +421,46 @@ TEST("on_error_return() replaces an error with a value") {
   SECTION("on_error_return() replaces an error with a value from the handler") {
     auto return_42 = [](const error&) { return expected<int>{42}; };
     SECTION("blueprint") {
-      check_eq(collect(range(1, 0)
-                         .concat(obs_error(sec::runtime_error))
-                         .on_error_return(return_42)),
+      check_eq(collect(
+                 range(1, 0).concat(obs_error()).on_error_return(return_42)),
```

Comment:
I think that we can refactor it to as we don't need to use range(1, 0) to create an empty observable anymore.
```suggestion
                 check_eq(collect(obs_error().on_error_return(return_42)), vector{42});
```